student-store-ui/src/components/ProductDetail/ProductDetail.jsx

```diff
@@ -1,10 +1,29 @@
 import * as React from "react"
+import axios from "axios";
 import { useParams } from "react-router-dom"
 import "./ProductDetail.css"
 
-export default function ProductDetail({ allProducts, handleAddItemToCart, handleRemoveItemToCart }) {
+export default function ProductDetail({ allProducts, handleAddItemToCart, handleRemoveItemToCart, setError }) {
     const { productId } = useParams();
-    const product = allProducts.find((product) => product.id === parseInt(productId))
+    const [isFetching, setIsFetching] = React.useState(true);
```

Comment:
I would initialize this as `false` and set it to `true` when fetch data is called since there's actually a time when the function is called and before useEffect is called that there is no fetching.